include/envoy/access_log/access_log.h

```diff
@@ -151,6 +151,11 @@ class RequestInfo {
   virtual Upstream::HostDescriptionConstSharedPtr upstreamHost() const PURE;
 
   /**
+   * Get the upstream local address, eg the source ip:port of the connection.
+   */
+  virtual const Optional<std::string>& upstreamLocalAddress() const PURE;
```

Comment:
Can we have this return an Address pointer? I'm going to need this for proto logging. I will also need to do something similar for downstream address but that is more complicated due to XFF so I can deal with that later, but this one should be easy to change.

Reply:
That will require changing the interface on Connection, which currently returns a reference, and needs to return a shared_ptr.  But I can do that if you'd like.

Reply:
Yeah please change if you don't mind.

Reply:
On second thought, the downstream address is already handled as a string.  Let's make a separate change to handle both of those as an Address::Instance, and change the Connection return type for remoteAddress() and localAddress().

Reply:
OK that's fine I will take care of it when I do the proto stuff.